### PR TITLE
Input validation functions jimmy

### PR DESCRIPTION
**Pull request by JimTK16**
I have added the validation into the save() function. 
The rules are:
- Task Name cannot be empty or longer than 40 chars.
- Task Description can be empty but no longer than 100 chars.

I also modified the app title in the index.html to 'Daily Tasks App' instead of 'Document'.

Note: on naming variables, I think it will be clearer if we add 'El' (Element) at the end of the variables' name. So instead of 'taskName', we can use 'taskNameEl' and the 'taskName' will be used for the actual value inside the taskName Element. Like:
const taskName = taskNameEl.value

Jimmy

### CODE REVIEW

**Review by andreimaier — COMMENTED** [review]
Could you maybe put the code inside a separate function and then call that function inside `function saveTask(e)` ?

File: public/index.js [hunk 1]
```diff
@@ -88,52 +86,80 @@ const date = document.querySelector('#date')
...
+    taskNameErrorEl.style.visibility = 'visible'
+    formHasError = true
+  } else if (taskName.trim().length > 40) {
+    taskNameErrorEl.innerHTML = 'Task Name must be under 40 characters'
+    taskNameErrorEl.style.visibility = 'visible'
+    formHasError = true
+  }
+
+  if (taskDescription.trim().length > 100) {
+    taskDescriptionErrorEl.innerHTML =
```
Comment (andreimaier) [inline public/index.js]: maybe use `.textContent` instead? Just generally safer than `innerHTML` or `innerText` I think

**Review by andreimaier — CHANGES_REQUESTED** [review]
I like the idea of using `taskNameEl` for variables that match HTML elements but not sure we need to declare another variable instead of just saying `.value`. But I can try to follow this convention if you want ! 😺 

- Please organize the code in a separate function
- Please look into sanitizing the content. I'll send you some ideas on discord
- see if you want to change `.innerHTML` to `.textContent`

**Review by jeru7 — APPROVED** [review]
I approved this since I don't see any problem. But just like what andrei said, you can use textContent or maybe innerText for safety purposes. Though ig you can stick with innerHtml since it's just for error messages but it will be a good practice to use textContent.

**Comment by andreimaier** [comment]
> I approved this since I don't see any problem. But just like what andrei said, you can use textContent or maybe innerText for safety purposes. Though ig you can stick with innerHtml since it's just for error messages but it will be a good practice to use textContent.

Yes, Like @jeru7 said, here it's not absolutely neccessary to stay away from innerHTML. But please think about sanitizing input and move the code into a separate function, `saveTask()` is getting bloated already

**Comment by JimTK16** [comment]
I have attempted to use DOMPurify to sanitize user input. I'm not sure i'm doing it the correct way though. I did some reading and also used Bing AI. To use DOMPurify, i added a <script> tag in index.html and use the sanitize() function in index.js.

**Review by andreimaier — APPROVED** [review]

**Comment by andreimaier** [comment]
The sanitizer you implemented works perfectly, good work! 

You can check it with dirty HTML ... input **** into the input and you'll see in the console that it removes some part of it 

Please resolve the merge conflicts and then merge your branch!
If you're not sure how to do it, maybe @jeru7 can help you, he's the ***Merge Conflict Master Solver***

**Pull request merged**